lib/actions/user.js

```diff
@@ -165,21 +199,66 @@ export function createOrUpdateUserMonitoredTrip (tripData, isNew, silentOnSucces
  */
-export function deleteUserMonitoredTrip (id) {
+export function deleteUserMonitoredTrip (tripId) {
   return async function (dispatch, getState) {
-    const { otp, user } = getState()
-    const { otp_middleware: otpMiddleware = null } = otp.config.persistence
-
-    if (otpMiddleware) {
-      const { accessToken } = user
-      const deleteResult = await deleteTrip(otpMiddleware, accessToken, id)
+    const { accessToken, apiBaseUrl, apiKey } = getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_MONITORTRIP_PATH}/${tripId}`
 
+    if (tripId) {
+      const deleteResult = await secureFetch(requestUrl, accessToken, apiKey, 'DELETE')
       if (deleteResult.status === 'success') {
         // Reload user's monitored trips after deletion.
         await dispatch(fetchUserMonitoredTrips(accessToken))
       } else {
         alert(`An error was encountered:\n${JSON.stringify(deleteResult)}`)
       }
+    } else {
+      alert('Corrupted state: Monitored Trip ID not available for exiting user.')
+    }
+  }
+}
+
+/**
+ * Requests a verification code via SMS for the logged-in user.
+ */
+export function requestPhoneVerificationSms (newPhoneNumber) {
+  return async function (dispatch, getState) {
+    const { accessToken, apiBaseUrl, apiKey, loggedInUser } = getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_USER_PATH}/${loggedInUser.id}${API_USER_VERIFYSMS_PATH}/${encodeURIComponent(newPhoneNumber)}`
+
+    const sendSmsResult = await secureFetch(requestUrl, accessToken, apiKey, 'GET')
+
+    if (sendSmsResult.status === 'success') {
+      // Refetch user and update application state with new phone number and verification status.
+      // (This also refetches the user's monitored trip, and that's ok.)
+      await dispatch(fetchOrInitializeUser({ accessToken }))
+    } else {
+      alert(`An error was encountered:\n${JSON.stringify(sendSmsResult)}`)
+    }
+  }
+}
+
+/**
+ * Validate the phone number verification code for the logged-in user.
+ */
+export function verifyPhoneNumber (code) {
+  return async function (dispatch, getState) {
+    const { accessToken, apiBaseUrl, apiKey, loggedInUser } = getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_USER_PATH}/${loggedInUser.id}${API_USER_VERIFYSMS_PATH}/${code}`
+
+    const sendResult = await secureFetch(requestUrl, accessToken, apiKey, 'POST')
+
+    // If the check is successful, status in the returned data will be "approved".
+    if (sendResult.status === 'success' && sendResult.data) {
+      if (sendResult.data.status === 'approved') {
+        // Refetch user and update application state with new phone number and verification status.
+        // (This also refetches the user's monitored trip, and that's ok.)
+        await dispatch(fetchOrInitializeUser({ accessToken }))
+      } else {
+        alert('You entered in incorrect validation code. Please try again.')
```

Comment:
Instead of saying "incorrect" maybe "invalid" would be a better catch-all for cases where the code is correct, but expired.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/708f5b7b8eb3f91aec836eaf0391b1c442a20bbb